internal/resource/route.go

```diff
@@ -638,6 +684,22 @@ func init() {
 					},
 				},
 			},
+			{
+				Title: ExpressionRouteRuleTitle,
+				Description: "When 'expression' is defined, 'priority' is required and " +
+					"'snis', 'sources' or 'destinations' cannot be set.",
+				If: &generator.Schema{
+					Required: []string{"expression"},
+				},
```

Comment:
Let's split this out into two different rules.
- priority is required
- all other fiends cannot be set.


Question: are host, paths, methods and headers allowed to be set when expression is used?

Reply:
done.

some of the other fields were hard to check because they were set by the ProcessDefaults method, so i updated that too.